### ISSUE TITLE
Unsupported rust generated (hex float literal unsupported)

### ISSUE THREAD

**Issue by pmatos** [jameysharp/corrode#73]
Corrode generated some code from a csmith file which is not supported in rust.

Compile with:
`[0]  pmatos@andromeda.desktop:~/Projects/corrode
(2) $ stack exec -- corrode -Wall -I/home/pmatos/Projects/csmith/runtime /home/pmatos/tmp/corrode-csmith/csmith_1.c
/home/pmatos/tmp/corrode-csmith/csmith_1.rs
/home/pmatos/tmp/corrode-csmith/csmith_1.o
 [0]  pmatos@andromeda.desktop:~/Projects/corrode
(3) $ rustc /home/pmatos/tmp/corrode-csmith/csmith_1.rs
/home/pmatos/tmp/corrode-csmith/csmith_1.rs:693:12: 693:17 error: hexadecimal float literal is not supported
/home/pmatos/tmp/corrode-csmith/csmith_1.rs:693            0x1.0p-100f32 * sf1 * (0x1.0p-28f32 * sf2)
                                                           ^~~~~
/home/pmatos/tmp/corrode-csmith/csmith_1.rs:693:35: 693:40 error: hexadecimal float literal is not supported
/home/pmatos/tmp/corrode-csmith/csmith_1.rs:693            0x1.0p-100f32 * sf1 * (0x1.0p-28f32 * sf2)
                                                                                  ^~~~~
error: aborting due to previous error`

[files.zip](https://github.com/jameysharp/corrode/files/489702/files.zip)

**Comment by jameysharp** [comment]
Yeah, I think I left a TODO comment in Corrode's translation of floating-point literals to this effect.

My understanding is that hex float literals are useful for precisely specifying the value you want, where decimal flat literals might lead to rounding error or something. So it seems worthwhile to open an issue on rustc asking for language support for this feature.

That said: have a look at what options I pass to csmith, and what macros I define when compiling its output. You'll find both in `scripts/csmith-test`. I haven't seen this particular bug in my testing, so perhaps you're seeing it because you're using different options than I am?
